src/microbe_stage/PlayerHoverInfo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   A system that manages reading what the player is hovering over.
+/// </summary>
+public class PlayerHoverInfo : Node
+{
+    private MicrobeCamera camera;
+    private CompoundCloudSystem cloudSystem;
+
+    /// <summary>
+    ///   All compounds the user is hovering over.
+    /// </summary>
+    public Dictionary<Compound, float> HoveredCompounds { get; private set; } = new Dictionary<Compound, float>();
+
+    /// <summary>
+    ///   All microbes the user is hovering over.
+    /// </summary>
+    public List<Microbe> HoveredMicrobes { get; private set; } = new List<Microbe>();
+
+    public bool IsHoveringAnyEntity => HoveredCompounds.Count > 0 || HoveredMicrobes.Count > 0;
+
+    public void Init(MicrobeCamera camera, CompoundCloudSystem cloudSystem)
+    {
+        this.camera = camera;
+        this.cloudSystem = cloudSystem;
+    }
+
+    public override void _Process(float delta)
+    {
+        HoveredCompounds = cloudSystem.GetAllAvailableAt(camera.CursorWorldPos);
+
+        var microbes = GetTree().GetNodesInGroup(Constants.AI_TAG_MICROBE);
+
+        foreach (var microbe in HoveredMicrobes)
+            microbe.IsHoveredOver = false;
+
+        HoveredMicrobes.Clear();
+
+        foreach (Microbe entry in microbes)
```

Comment:
I feel like entry is too generic a name. As the loop is short, this is minor, but it could still benefit a name change, I think.

Reply:
my suggestion: `foreach (var microbe in microbes)`

I think this might be some of my first C# code that I wrote for Thrive, that's why it is pretty rough...

Reply:
Yeah, entry is a very common var name in foreach loops in the project, I actually supposed it was some automation result. Could use some refactor, perhaps.

I second this choice, but note the loop right above also uses `microbe` for vars looped from another set.

Reply:
If it's already used, maybe both loop variables need renaming to reflect their relationship with each other...

Reply:
The loop above loops through microbes in HoveredMicrobes list (to clear them properly from before update), so I'd suggest `hoveredMicrobe` and `worldMicrobe`. But this is first-thought material, so if anyone comes with a better idea I'm all open.

Reply:
Maybe `hoveredMicrobe` for the hovered ones and `microbe` for world microbes, also I'll probably rename `microbes` list to `allMicrobes`.